vars/getJobTimeouts.groovy

```diff
@@ -4,26 +4,38 @@ List<Integer> call(Map params, String region){
     // handle params which can be a json list
     def test_config = groovy.json.JsonOutput.toJson(params.test_config)
     def cmd = """
-    #!/bin/bash
-    export SCT_CLUSTER_BACKEND="${params.backend}"
-    export SCT_CONFIG_FILES=${test_config}
-    if [[ -n "${params.region ? params.region : ''}" ]] ; then
-        export SCT_REGION_NAME=${groovy.json.JsonOutput.toJson(params.region)}
-    fi
+        #!/bin/bash
+        export SCT_CLUSTER_BACKEND="${params.backend}"
+        export SCT_CONFIG_FILES=${test_config}
+        if [[ -n "${params.region ? params.region : ''}" ]] ; then
+            export SCT_REGION_NAME=${groovy.json.JsonOutput.toJson(params.region)}
+        fi
 
-    if [[ -n "${params.gce_datacenter ? params.gce_datacenter : ''}" ]] ; then
-        export SCT_GCE_DATACENTER=${groovy.json.JsonOutput.toJson(params.gce_datacenter)}
-    fi
+        if [[ -n "${params.gce_datacenter ? params.gce_datacenter : ''}" ]] ; then
+            export SCT_GCE_DATACENTER=${groovy.json.JsonOutput.toJson(params.gce_datacenter)}
+        fi
 
-    if [[ -n "${params.azure_region_name ? params.azure_region_name : ''}" ]] ; then
-        export SCT_AZURE_REGION_NAME=${groovy.json.JsonOutput.toJson(params.azure_region_name)}
-    fi
-    ./docker/env/hydra.sh output-conf -b "${params.backend}"
-    """
+        if [[ -n "${params.azure_region_name ? params.azure_region_name : ''}" ]] ; then
+            export SCT_AZURE_REGION_NAME=${groovy.json.JsonOutput.toJson(params.azure_region_name)}
+        fi
+        ./docker/env/hydra.sh output-conf -b "${params.backend}"
+        """
     def testData = sh(script: cmd, returnStdout: true).trim()
     println(testData)
-    testData = testData =~ /test_duration: (\d+)/
-    testDuration = testData[0][1].toInteger()
+    if (params.stress_duration == "") {
+        testData = testData =~ /test_duration: (\d+)/
+        testDuration = testData[0][1].toInteger()
+    } else {
+        stressDuration = params.stress_duration.toInteger()
+        try {
+            prepareDuration = params.prepare_duration.toInteger()
+        } catch (e) {
+            testData = testData =~ /prepare_duration: (\d+)/
+            prepareDuration = testData[0][1].toInteger()
+        }
+        Integer stressEndup = 10
```

Comment:
10min ? I think it need at least 30min, which is the max time we wait for nemesis to finish

Reply:
it is time to wait while stress command wait to finished. It is configured in hour sct code for stress commands

Reply:
o.k.